app/controllers/listings_controller.rb

```diff
@@ -1,5 +1,14 @@
 class ListingsController < ApplicationController
   def index
-    @listings = Listing.all
+    @listings = Listing.where(nil)
+    filtering_params(params).each do |key, value|
+      @listings = @listings.public_send(key, value) if value.present?
+    end
+  end
+
+private
+
+  def filtering_params(params)
+    params.slice(:city, :max_guests, :category_id, :nightly_rate)
```

Comment:
slice(*keys) public
Slice a hash to include only the given keys. This is useful for limiting an options hash to valid keys before passing to a method: